src/Microsoft.VisualStudio.Composition/Reflection/TypeRef.cs

```diff
@@ -106,6 +117,15 @@ private TypeRef(Resolver resolver, Type type)
 
         public ImmutableArray<TypeRef> GenericTypeArguments { get; private set; }
 
+        /// <summary>
+        /// Gets the full list of base types and interfaces for this instance.
+        /// This list will only be populated if this instance was created with shallow set to false.
```

Comment:
>This list will only be populated if this instance was created with shallow set to false. [](start = 12, length = 88)

What will it be if `shallow` is true? Will it be empty or "uninitialized" (as only an `ImmutableArray<T>` can be). How can I tell if this is the case? Is there a `IsShallow` property or something I can check?

Reply:
There is currently no way to determine if a `TypeRef` is shallow. Generally speaking the `TypeRef`s you get ahold of will not be shallow since they will be at the bottom of the type hierarchy.

Reply:
Do non-shallow objects *ever* have an *empty* array here? If not, then an empty array could be an indicator and could be documented (or used to implement a property getter: `internal bool IsShallow => this.BaseTypes.IsEmpty`.) Do value types and reference types both have base types that are added to this array?

---
In reply to: [266672285](https://github.com/Microsoft/vs-mef/pull/111#discussion_r266672285) [](ancestors = 266672285)

Reply:
Non shallow objects can have an empty array. `object` is one example. This can also happen with TypeRefs to interfaces.